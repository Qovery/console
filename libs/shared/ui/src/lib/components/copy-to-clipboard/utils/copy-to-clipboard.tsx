export function copyToClipboard(str: string): void {
  navigator.clipboard.writeText(str)
}
