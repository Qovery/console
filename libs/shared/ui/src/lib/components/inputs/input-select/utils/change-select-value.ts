import { act, fireEvent, getByRole, getByTestId } from '@testing-library/react'

export function simulateSpaceKeyClick(element: Element) {
  fireEvent.keyDown(element, { key: ' ' })
  fireEvent.keyUp(element, { key: ' ' })
}

export async function changeSelectValueByLabel(
  baseElement: HTMLElement,
  selectTestId: string,
  value: string
): Promise<void> {
  simulateSpaceKeyClick(getByTestId(getByTestId(baseElement, selectTestId), 'input-select-button'))

  const itemToSelect = getByRole(baseElement, 'option', {
    name: value,
  })
  await act(() => {
    fireEvent.click(itemToSelect)
  })
}
