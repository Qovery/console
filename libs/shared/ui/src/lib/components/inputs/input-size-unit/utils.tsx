export const inputSizeUnitRules = (maxSize?: number, minSize = 0) => ({
  required: 'Please enter a size.',
  validate: (value: number) => (maxSize ? value <= maxSize : undefined),
  max: maxSize,
  min: minSize,
  pattern: {
    value: /^[0-9]+$/,
    message: 'Please enter a number.',
  },
})
