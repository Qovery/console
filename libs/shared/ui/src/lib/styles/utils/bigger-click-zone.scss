.bigger-click-zone {
  position: relative;

  &:after {
    content: '';
    position: absolute;
    inset: -4px;
    // background: rgba(red, .3);
    display: block;
  }
}
