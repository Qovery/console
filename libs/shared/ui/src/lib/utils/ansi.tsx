// eslint-disable-next-line @typescript-eslint/no-restricted-imports
import AnsiPrimitive from 'ansi-to-react'
import { type ComponentProps } from 'react'
import { twMerge } from '@qovery/shared/util-js'

interface AnsiProps extends ComponentProps<typeof AnsiPrimitive> {}

export function Ansi({ className, children, ...props }: AnsiProps) {
  return (
    <AnsiPrimitive className={twMerge('code-ansi', className)} linkify {...props}>
      {children}
    </AnsiPrimitive>
  )
}
