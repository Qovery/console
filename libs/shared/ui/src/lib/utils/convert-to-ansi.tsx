import Ansi from 'ansi-to-react'
import { ReactNode } from 'react'

export const convertToAnsi = (content = ''): ReactNode => {
  return (
    <Ansi className="code-ansi" linkify>
      {content}
    </Ansi>
  )
}
