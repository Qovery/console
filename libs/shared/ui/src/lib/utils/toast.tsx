import { Toast, toast as toastAction } from 'react-hot-toast'
import { ToastContent } from '../components/toast/toast'

export enum ToastEnum {
  SUCCESS = 'SUCCESS',
  ERROR = 'ERROR',
  WARNING = 'WARNING',
}

export const toast = (
  status: ToastEnum,
  title: string,
  description?: string,
  callback?: () => void,
  iconAction?: string,
  labelAction?: string,
  externalLink?: string
) => {
  return toastAction.success((options: Toast) =>
    ToastContent(status, options, title, description, callback, iconAction, labelAction, externalLink)
  )
}

export default toast
