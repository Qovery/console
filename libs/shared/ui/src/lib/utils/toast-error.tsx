import { type SerializedError } from '@qovery/shared/utils'
import toast, { ToastEnum } from './toast'

export function toastError(error: SerializedError | Error, title?: string, message?: string): void {
  toast(ToastEnum.ERROR, title || error.name || 'Error', message || error.message || 'No message found')
}
