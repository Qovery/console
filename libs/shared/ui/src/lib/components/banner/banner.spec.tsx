import { act, fireEvent, getByRole, getByTestId } from '@testing-library/react'
import { render } from '__tests__/utils/setup-jest'
import Banner, { BannerStyle } from './banner'

describe('Banner', () => {
  it('should render successfully', () => {
    const { baseElement } = render(<Banner bannerStyle={BannerStyle.WARNING}>Hello</Banner>)
    expect(baseElement).toBeTruthy()
  })

  it('should render in orange', () => {
    const { baseElement } = render(<Banner bannerStyle={BannerStyle.WARNING}>Hello</Banner>)
    const banner = getByTestId(baseElement, 'banner')
    expect(banner).toHaveClass('bg-warning-500')
    expect(banner).toHaveClass('text-warning-900')
  })

  it('should render in brand color', () => {
    const { baseElement } = render(<Banner bannerStyle={BannerStyle.PRIMARY}>Hello</Banner>)
    const banner = getByTestId(baseElement, 'banner')
    expect(banner).toHaveClass('bg-brand-500')
    expect(banner).toHaveClass('text-white')
  })

  it('should render a button and handle click', async () => {
    const onClickSpy = jest.fn()
    const { baseElement } = render(
      <Banner bannerStyle={BannerStyle.PRIMARY} onClickButton={onClickSpy} buttonLabel="Click me!">
        Hello
      </Banner>
    )
    const button = getByRole(baseElement, 'button', { name: 'Click me!' })

    await act(() => {
      fireEvent.click(button)
    })

    expect(onClickSpy).toHaveBeenCalled()
  })
})
