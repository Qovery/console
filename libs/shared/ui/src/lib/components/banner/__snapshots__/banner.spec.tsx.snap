// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Banner should render in brand color 1`] = `
<body>
  <div>
    <div
      class="flex h-10 items-center justify-center text-sm font-medium bg-brand-500 text-white"
    >
      Hello
    </div>
  </div>
</body>
`;

exports[`Banner should render in yellow 1`] = `
<body>
  <div>
    <div
      class="flex h-10 items-center justify-center text-sm font-medium bg-yellow-500 text-yellow-900"
    >
      Hello
    </div>
  </div>
</body>
`;
